adding new gui file

nahid-tanjum opened the pull request:
i add logic to handle 3line and 2line TLE file in sat_sim_handler and implemented header in output file so that it will print the number of satellite and the name of all satellite. now the timestep in minutes. also i add gui in json file. add gui argument in cli arg under sat_sim_arg

nahid-tanjum committed:
adding new gui file

i add logic to handle 3line and 2line TLE file in sat_sim_handler and implemented
header in output file so that it will print the number of satellite and the name of all satellite. now the timestep in minutes. also i add gui in json file. add gui argument in cli arg under sat_sim_arg

nahid-tanjum committed:
fix output functionality

restore output_to_file in config.
added output_to_file attribute and set_output_to_file method in sat_sim.
modified run_with_adj_martix to handle the output.
imported utc from skyfield.api  in both config and core.
modified _convert_to_datetime method to utc.

nahid-tanjum committed:
upload txt and csv file and modify satsim in line 32 to avoid boolean Evaluation

generate txt and csv file for algo block.
Modify the Condition to Avoid Boolean Evaluation of Time Objects in line 32.

nahid-tanjum committed:
add data class and remove comma from txt file and modified txt format so that it's not append

add data class in output file
remove comma from txt file.
modified code so that txt file not append.

nahid-tanjum committed:
Changed Satellite Labels to Be Separated by Commas

nahid-tanjum committed:
removed timestep as a separate variable

removed timestep as a separate variable in output dataclass

nahid-tanjum committed:
modify save_martices method to set self.keys and adjust get_result method so that self.keys is populated before returning the result

Modify the save_matrices method to set self.keys even when the output_file_type is set to 'false'
Ensure that self.keys is populated before returning the result.

nahid-tanjum committed:
modified output file name so that it generate the output file based on the simulation local time and modified output file path so that output file save under sat_sim_output folder

modified output file name so that it generate the output file based on the simulation local time
modified output file path so that output file save under sat_sim_output folder
remove redundant option from json file

elysiacodes reviewed (changes requested):
As discussed in Discord, have the sat sim output files be unique with a date/time in its filename to an output folder inside sat_sim :)

nahid-tanjum committed:
remove matrix size from csv file

remove matrix size from each of the matrix from csv file as we have number of satellite at the output header
